src/app/models/field/baseField.ts

```diff
@@ -22,14 +22,24 @@ export const MyInfoSchema = new Schema<IMyInfoSchema>(
       enum: Object.values(MyInfoAttribute),
       validate: {
         validator: function (this: IMyInfoSchema) {
-          const { authType, responseMode } = this.ownerDocument()
+          const form = this.ownerDocument()
+          const { authType, responseMode } = form
+          let myInfoCount = 0
+          if (form.form_fields !== undefined) {
+            form.form_fields.forEach((field) => {
+              if (field.myInfo !== undefined) {
+                myInfoCount++
+              }
+            })
+          }
           return (
             authType === AuthType.MyInfo &&
-            responseMode !== ResponseMode.Encrypt
+            responseMode !== ResponseMode.Encrypt &&
+            myInfoCount <= 30
           )
```

Comment:
I'm wondering whether we can improve this chunk of code by moving it into the validation for the Form schema. I think that's where this code really belongs, as this is really form-level validation ("every form should have no more than 30 MyInfo fields"), not field-level validation. This would have the added advantages of 1) running the validator only once per form, rather than once per field and 2) removing the need for breaking encapsulation by calling `this.ownerDocument()`. what do you think?

Reply:
Makes sense, moving this to the main form schema to validate before saving.